src/devices/HuskyLens/HuskyLens.cs

```diff
@@ -0,0 +1,149 @@
+        public HuskyLens(IBinaryConnection connection)
+        {
+            _connection = connection;
+            Algorithm = Algorithm.Undefined;
+        }
+
+        /// <summary>
+        /// Test connection with HuskyLens.
+        /// </summary>
+        /// <returns>true for success, guess the rest</returns>
+        public bool Ping()
+        {
+            // COMMAND_REQUEST_KNOCK(0x2C):
+            // Used for test connection with HUSKYLENS.When HUSKYLENS received this command, HUSKYLENS will return COMMAND_RETURN_OK
+            // See: https://github.com/HuskyLens/HUSKYLENSArduino/blob/master/HUSKYLENS%20Protocol.md#command_request_knock0x2c
+            _connection.Write(new byte[] { 0x55, 0xAA, 0x11, 0x00, 0x2C, 0x3C });
+            WaitForOK();
+            return true;
+        }
+
+        /// <summary>
+        /// Gets all recognized objects
+        /// </summary>
+        public IReadOnlyCollection<HuskyObject> GetAllObjects()
+        {
+            // COMMAND_REQUEST (0x20):
+            // Request all blocks and arrows from HUSKYLENS.
+            // See: https://github.com/HuskyLens/HUSKYLENSArduino/blob/master/HUSKYLENS%20Protocol.md#command_request-0x20
+            var command = new byte[] { 0x55, 0xAA, 0x11, 0x00, 0x20, 0x30 };
+            _connection.Write(command);
+
+            // wait for COMMAND_RETURN_INFO
+            var response = new DataFrame(_connection.Read(0x0A + 6));
+            if (!response.IsValid() || response.Command != Command.COMMAND_RETURN_INFO)
+            {
+                // error
+                throw new NotImplementedException();
+            }
+
+            // read # of blocks&arrows
+            var count = BinaryPrimitives.ReadInt16LittleEndian(response.Data.Slice(0, 2));
+            Console.WriteLine($"Reading {count} objects");
+            var huskyObjects = new List<HuskyObject>();
+            for (int i = 0; i < count; i++)
+            {
+                response = new DataFrame(_connection.Read(0x0A + 6));
+                if (!response.IsValid())
+                {
+                    // error
+                    throw new NotImplementedException();
+                }
+
+                switch (response.Command)
+                {
+                    case Command.COMMAND_RETURN_BLOCK:
+                        huskyObjects.Add(Block.FromData(response.Data));
+                        break;
+                    case Command.COMMAND_RETURN_ARROW:
+                        huskyObjects.Add(Arrow.FromData(response.Data));
+                        break;
+                    default:
+                        // error
+                        throw new NotImplementedException();
+                }
+            }
+
+            return huskyObjects;
+        }
+
+        private void WaitForOK()
+        {
+            var response = _connection.Read(6);
+
+            if (!IsOk(response))
+            {
+                throw new Exception();
```

Comment:
please add message and change error type to something meaningful i.e. InvalidOperationException. If it's actionable (i.e. please ensure device is connected properly or something else) then please also add message